src/model/mod.rs

```diff
@@ -822,6 +824,87 @@ pub fn match_costs(games: &[Game]) -> Option<Vec<MatchCost>> {
     Some(match_costs)
 }
 
+fn game_win_record(game: &Game) -> GameWinRecord {
+    let game_id = game.id;
+    let (winners, losers, winner_team, loser_team) = identify_game_winners_losers(game);
+
+    GameWinRecord {
+        game_id,
+        winners,
+        losers,
+        winner_team,
+        loser_team
+    }
+}
+
+/// Identifies the winners and losers of a game.
+/// Return format is tuple of (winner ids, loser ids, winner team, loser team)
+fn identify_game_winners_losers(game: &Game) -> (Vec<i32>, Vec<i32>, i32, i32) {
+    if game.team_type == TeamType::HeadToHead {
+        if game.match_scores.len() != 2 {
+            panic!("Head to head game must have 2 players: {:?}", game);
+        }
+
+        // Head to head
+        let score_0 = game.match_scores.index(0);
+        let score_1 = game.match_scores.index(1);
+
+        let head_to_head = score_0.team == 0 && score_1.team == 1;
+
+        if head_to_head {
+            let winners;
+            let losers;
+
+            if score_0.score > score_1.score {
+                winners = vec![score_0.player_id];
+                losers = vec![score_1.player_id];
+            }
+            else {
+                winners = vec![score_1.player_id];
+                losers = vec![score_0.player_id];
+            }
+
+            return (winners, losers, 0, 0);
+        }
+    }
+
+    if game.team_type != TeamType::TeamVs {
+        panic!("Invalid team type: {:?}", game);
+    }
+
+    let mut red_players = vec![];
+    let mut blue_players = vec![];
+
+    let mut red_scores: Vec<i64> = vec![];
+    let mut blue_scores: Vec<i64> = vec![];
+
+    let red = 2;
+    let blue = 1;
+
+    for score in &game.match_scores {
+        if score.team == red {
+            red_players.push(score.player_id);
+            red_scores.push(score.score);
+        }
+        else if score.team == blue {
+            blue_players.push(score.player_id);
+            blue_scores.push(score.score);
+        }
+        else {
+            panic!("Invalid team type");
+        }
+    }
```

Comment:
matching is slightly cleaner + would be even cleaner if team types are ever changed into an enum

```rust
for score in &game.match_scores {
	match score.team {
		i if i == blue => {
			blue_players.push(score.player_id);
			blue_scores.push(score.score);
		},
		i if i == red => {
			red_players.push(score.player_id);
			red_scores.push(score.score);
		},
		_ => panic!("Invalid team type")
	}
}
```

Reply:
Sorry I am a C# dev as you can see